scrunch/tests/test_datasets.py

```diff
@@ -250,6 +256,17 @@ def test_edit_dataset(self):
     def process_expr_side_effect(self, expr, ds):
         return expr
 
+    def test_replace_from_csv(self):
+        ds_mock = self._dataset_mock()
+        ds = StreamingDataset(ds_mock)
+        assert ds.resource.body.get('streaming') == 'streaming'
+        file = StringIO()
+        file.write("id, age\n1, 15")
+        file.seek(0)
+        with pytest.raises(AttributeError):
+            assert ds.replace_from_csv(file, push_rows=False) == 1
+        assert ds.resource.body.get('streaming') == 'streaming'
```

Comment:
Also assert that the rows have been streamed in, or mock that the `push_rows` function was called with the rows you wanted`. Right now this test isn't protecting the implementation.